pkg/dependency/parser/python/uv/parse.go

```diff
@@ -0,0 +1,166 @@
+package uv
+
+import (
+	"sort"
+
+	"github.com/BurntSushi/toml"
+	"github.com/samber/lo"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/dependency"
+	ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+	xio "github.com/aquasecurity/trivy/pkg/x/io"
+)
+
+type Lock struct {
+	Packages []Package `toml:"package"`
+}
+
+func (l Lock) packages() map[string]Package {
+	return lo.SliceToMap(l.Packages, func(pkg Package) (string, Package) {
+		return pkg.Name, pkg
+	})
+}
+
+func (l Lock) directDeps() map[string]struct{} {
+	deps := make(map[string]struct{})
+	root := l.root()
+	for _, dep := range root.Dependencies {
+		deps[dep.Name] = struct{}{}
+	}
+	return deps
+}
+
+func (l Lock) prodDeps() map[string]struct{} {
+	packages := l.packages()
+	visited := make(map[string]struct{})
+	walkPackageDeps(l.root(), packages, visited)
+	return visited
+}
+
+func walkPackageDeps(pkg Package, packages map[string]Package, visited map[string]struct{}) {
+	if _, ok := visited[pkg.Name]; ok {
+		return
+	}
+	visited[pkg.Name] = struct{}{}
+	for _, dep := range pkg.Dependencies {
+		depPkg, exists := packages[dep.Name]
+		if !exists {
+			continue
+		}
+		walkPackageDeps(depPkg, packages, visited)
+	}
+}
+
+func (l Lock) root() Package {
+	for _, pkg := range l.Packages {
+		if pkg.isRoot() {
+			return pkg
+		}
+	}
+
+	return Package{}
+}
+
+type Package struct {
+	Name         string       `toml:"name"`
+	Version      string       `toml:"version"`
+	Source       Source       `toml:"source"`
+	Dependencies []Dependency `toml:"dependencies"`
+}
+
+// https://github.com/astral-sh/uv/blob/f7d647e81d7e1e3be189324b06024ed2057168e6/crates/uv-resolver/src/lock/mod.rs#L572-L579
+func (p Package) isRoot() bool {
+	return p.Source.Editable == "." || p.Source.Virtual == "."
+}
+
+type Source struct {
+	Editable string `toml:"editable"`
+	Virtual  string `toml:"virtual"`
+}
+
+type Dependency struct {
+	Name string `toml:"name"`
+}
+
+type Parser struct{}
+
+func NewParser() *Parser {
+	return &Parser{}
+}
+
+func (p *Parser) Parse(r xio.ReadSeekerAt) ([]ftypes.Package, []ftypes.Dependency, error) {
+	var lock Lock
+	if _, err := toml.NewDecoder(r).Decode(&lock); err != nil {
+		return nil, nil, xerrors.Errorf("failed to decode uv lock file: %w", err)
+	}
+
+	rootPackage := lock.root()
+	if rootPackage.Name == "" {
+		return nil, nil, xerrors.New("uv lockfile does not contain a root package. " +
+			"See https://github.com/astral-sh/uv/blob/f80ddf10b63c3e7b421ca4658e63f97db1e0378c/crates/uv/src/commands/project/lock.rs#L933-L936")
```

Comment:
I think it makes no sense to share the link for users.
```suggestion
	// lock file should include root package
	// cf. https://github.com/astral-sh/uv/blob/f80ddf10b63c3e7b421ca4658e63f97db1e0378c/crates/uv/src/commands/project/lock.rs#L933-L936
	if rootPackage.Name == "" {
		return nil, nil, xerrors.Errorf("unable to find root package for uv.lock file.")
	}
```